pkg/azuredisk/controllerserver.go

```diff
@@ -543,30 +543,54 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	if err != nil {
 		return nil, status.Error(codes.Internal, fmt.Sprintf("Unknown list snapshot error: %v", err))
 	}
+
+	start := 0
+	if req.StartingToken != "" {
+		start, err = strconv.Atoi(req.StartingToken)
+		if err != nil {
+			return nil, err
+		}
+		if start >= len(snapshotListPage.Values()) {
+			return nil, status.Errorf(codes.Aborted, "ListSnapshots starting token(%d) is greater than total number of snapshots", start)
+		}
+	}
+	perPage := 0
+	if req.MaxEntries > 0 {
+		perPage = int(req.MaxEntries)
+	}
+	nextToken := len(snapshotListPage.Values())
```

Comment:
move following code into last since `nextToken` is only used in the last return statement
```
	nextToken := len(snapshotListPage.Values())
	if start+perPage < len(snapshotListPage.Values()) {
		nextToken = start + perPage
	}
```

Reply:
done.